Author: Jasper Grimm (UoY)
Ensure that the python test environment extends LD_LIBRARY_PATH, PYTHONPATH
 instead of overwriting these
diff -Nru nlopt-2.6.2.orig/test/CMakeLists.txt nlopt-2.6.2/test/CMakeLists.txt
--- nlopt-2.6.2.orig/test/CMakeLists.txt	2021-05-21 10:15:32.000000000 +0100
+++ nlopt-2.6.2/test/CMakeLists.txt	2021-05-21 10:16:18.000000000 +0100
@@ -51,8 +51,8 @@
 endforeach ()
 
 if (NUMPY_FOUND AND PYTHONLIBS_FOUND AND (SWIG_FOUND OR (EXISTS ${PROJECT_SOURCE_DIR}/src/swig/nlopt-python.cpp)))
-  set (PYINSTALLCHECK_ENVIRONMENT "LD_LIBRARY_PATH=${PROJECT_BINARY_DIR}/src/swig"
-                                  "PYTHONPATH=${PROJECT_BINARY_DIR}/src/swig"
+  set (PYINSTALLCHECK_ENVIRONMENT "LD_LIBRARY_PATH=${PROJECT_BINARY_DIR}/src/swig:$ENV{LD_LIBRARY_PATH}"
+                                  "PYTHONPATH=${PROJECT_BINARY_DIR}/src/swig:$ENV{PYTHONPATH}"
     )
   add_test (NAME test_python COMMAND ${PYTHON_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/t_python.py)
   set_tests_properties (test_python PROPERTIES ENVIRONMENT "${PYINSTALLCHECK_ENVIRONMENT}")
